utilities/lint.sh

```diff
@@ -0,0 +1,19 @@
+#! /usr/bin/env bash
+set -eo pipefail
+
+ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")"/.. && pwd)"
+errors=0
+
+if ! make -C "${ROOT_DIR}/collector" check; then
+    ((errors = errors + 1))
+fi
+if ! make -C "${ROOT_DIR}" shfmt-check; then
+    ((errors = errors + 1))
+fi
+if ! make -C "${ROOT_DIR}" shellcheck-all; then
+    ((errors = errors + 1))
+fi
+
+if ((errors)); then
+    echo >&2 "${errors} linters have failed"
```

Comment:
can we also have some logging that describes which linters failed?

Reply:
I mean, we could, but the output will already have the linters output above this line, I'd prefer to not complicate the logic in this script to also be appending names onto a variable to be print, WDYT?